Add attachment support

Frank995 opened the pull request:
### Summary of changes
Add functions for attachments retrieval

## Checklist

- [ ] Changes represent a *discrete update*
- [x] Commit messages are meaningful and descriptive
- [ ] Changeset does not include any extraneous changes unrelated to the discrete change

Frank995 commented:
I'm open to critics regarding code standards and if you have some posts with attachments in the test discourse please share id and slug so that I can add tests

goetzk commented:
Also please add a "Closes #95" to your commit message

goetzk commented on src/pydiscourse/client.py:
```diff
@@ -604,6 +605,87 @@ def delete_topic(self, topic_id, **kwargs):
+
+    def topic_attachments_number(self, slug, topic_id, **kwargs):
+        """
+
```
Please add a comment explaining what this function does (for example see https://github.com/pydiscourse/pydiscourse/blob/53be265b2345b2a294c79ec4c35d40992197f3a8/src/pydiscourse/client.py#L1370)

goetzk commented on src/pydiscourse/client.py:
```diff
@@ -604,6 +605,87 @@ def delete_topic(self, topic_id, **kwargs):
+
+    def topic_attachments(self, slug, topic_id, allowed_extensions=None, **kwargs):
+        """
+
```
Please add a comment explaining what this function does (for example see https://github.com/pydiscourse/pydiscourse/blob/53be265b2345b2a294c79ec4c35d40992197f3a8/src/pydiscourse/client.py#L1370
In particular this method name gives no indication of if its for listing or uploading

goetzk commented on src/pydiscourse/client.py:
```diff
@@ -604,6 +605,87 @@ def delete_topic(self, topic_id, **kwargs):
+    
+    def _topic_attachments_urls(self, slug, topic_id, **kwargs):
+        """
+
```
Please add example comment, eg "Extract URLs of attachments from every post in topic"

goetzk commented on src/pydiscourse/client.py:
```diff
@@ -604,6 +605,87 @@ def delete_topic(self, topic_id, **kwargs):
+        posts = [post['cooked'] for post in \
+                 topic['post_stream']['posts']]
+        attachments_urls = [ATTACHMENTS_REGEX.findall(post) for post in posts]
+        attachments_urls = {
```
Question - why using the same variable name twice here?

goetzk commented:
Can I confirm the behaviour topic_attachments? (sorry to bounce this to you as a question).

- takes a topic, eg https://meta.discourse.org/t/understanding-uploads-images-and-attachments/275735
- uses the other methods to extract attachment urls, eg https://d11a6trkgmumsb.cloudfront.net/original/4X/b/5/d/b5d483135cb1405290f07bfe4a9f5bd6dd9b6a1b.png
- Checks the extension is in the permitted list

Have I got that right?

Thanks for the contribution - most of my feedback is questions or requests for short non functional changes but they'd be nice to have.

thanks,
kk

Frank995 commented:
> Can I confirm the behaviour topic_attachments? (sorry to bounce this to you as a question).
> 
>     * takes a topic, eg https://meta.discourse.org/t/understanding-uploads-images-and-attachments/275735
> 
>     * uses the other methods to extract attachment urls, eg https://d11a6trkgmumsb.cloudfront.net/original/4X/b/5/d/b5d483135cb1405290f07bfe4a9f5bd6dd9b6a1b.png
> 
>     * Checks the extension is in the permitted list
> 
> 
> Have I got that right?
> 
> Thanks for the contribution - most of my feedback is questions or requests for short non functional changes but they'd be nice to have.
> 
> thanks, kk

Hi sorry for the big delay, I've been very busy. Yes that would be the intended behaviour. I also addressed your comments, let me know if they make more sense to you know

Frank995 commented:
Hi, any news on this?

goetzk commented:
> Hi, any news on this?

Hi Frank,
Import logging looks like its the only extraneous move now (https://github.com/pydiscourse/pydiscourse/pull/96/files#diff-6cd9d84a11416e5456653817c28de5b33224ec63dfc5776b3e1b567454df0c75) so I'm hitting the button run the testsuite and we'll check our status after that runs.

goetzk commented:
@Frank995 it looks like the linter has a few complaints about whitespace issues 

https://github.com/pydiscourse/pydiscourse/actions/runs/7030837244/job/20000450106?pr=96

Could you please resolve them?

thanks,
Karl.

Frank995 commented:
> @Frank995 it looks like the linter has a few complaints about whitespace issues
> 
> https://github.com/pydiscourse/pydiscourse/actions/runs/7030837244/job/20000450106?pr=96
> 
> Could you please resolve them?
> 
> thanks, Karl.

Hi, I should have fixed the above mentioned problems. Let me know

goetzk commented:
> > @Frank995 it looks like the linter has a few complaints about whitespace issues
> > https://github.com/pydiscourse/pydiscourse/actions/runs/7030837244/job/20000450106?pr=96
> > Could you please resolve them?
> > thanks, Karl.
> 
> Hi, I should have fixed the above mentioned problems. Let me know

Sorry about the long turn arounds on this.

I can't merge yet because coverage is in client.py < 46% while the current coverage is 45% (yes really, 1% :/).

I will look at pushing that up a bit in the coming days.

thanks,
Karl.

Frank995 commented:
I had forgotten about this for a while. Any chance it gets merged?
